File: SupplierPortal.Web/Modules/ServerTypes/Portal/SupplierColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { SupplierRow } from "./SupplierRow";

export interface SupplierColumns {
    Id: Column<SupplierRow>;
    CompanyName: Column<SupplierRow>;
    Email: Column<SupplierRow>;
}

export class SupplierColumns extends ColumnsBase<SupplierRow> {
    static readonly columnsKey = 'Portal.Supplier';
    static readonly Fields = fieldsProxy<SupplierColumns>();
}

File: SupplierPortal.Web/Modules/Portal/Supplier/SupplierColumns.cs
﻿using Serenity.ComponentModel;
using System.ComponentModel;

namespace SupplierPortal.Portal.Columns;

[ColumnsScript("Portal.Supplier")]
[BasedOnRow(typeof(SupplierRow), CheckNames = true)]
public class SupplierColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string CompanyName { get; set; }
    public string Email { get; set; }
}